pravega/pravega-operator
# Pravega Operator API Resource Hook.

### Change log description

* Adds a manifest which defines a post-install/upgrade hook to make sure the `pravegacluster` resource exists.

### Purpose of the change

* If one is unlucky with the timings when install the `pravega-operator` chart, directly followed by the `pravega` chart, the api-resource `PravegaCluster` may not exist by the time the pravega chart tries to create a `PravegaCluster`.

### What the code does

* See above.

### How to verify it

Run many install/uninstall cycles of the mentioned charts -- see that the pravega chart does not start before its resource dependency is registered.
